src/generated/tokens.rs

```diff
@@ -0,0 +1,1060 @@
+
+impl ToRedisArgs for Skipme {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        "SKIPME".write_redis_args(out);
+        self.0.write_redis_args(out);
+    }
+}
+/// Redis Type: Enabled
+pub enum Enabled {
+    /// ON
+    On,
+    /// OFF
+    Off,
+}
+
+impl ToRedisArgs for Enabled {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        match self {
+            Enabled::On => "ON".write_redis_args(out),
+            Enabled::Off => "OFF".write_redis_args(out),
+        }
+    }
+}
+/// Redis Type: OnOffSkip
+pub enum OnOffSkip {
+    /// ON
+    On,
+    /// OFF
+    Off,
+    /// SKIP
+    Skip,
+}
+
+impl ToRedisArgs for OnOffSkip {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        match self {
+            OnOffSkip::On => "ON".write_redis_args(out),
+            OnOffSkip::Off => "OFF".write_redis_args(out),
+            OnOffSkip::Skip => "SKIP".write_redis_args(out),
+        }
+    }
+}
+/// Redis Type: Status
+pub enum Status {
+    /// ON
+    On,
+    /// OFF
+    Off,
+}
+
+impl ToRedisArgs for Status {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        match self {
+            Status::On => "ON".write_redis_args(out),
+            Status::Off => "OFF".write_redis_args(out),
+        }
+    }
+}
+/// Redis Type: REDIRECT
+pub struct Redirect(i64);
+
+impl ToRedisArgs for Redirect {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        "REDIRECT".write_redis_args(out);
+        self.0.write_redis_args(out);
+    }
+}
+/// Redis Type: PREFIX
+pub struct Prefix(String);
+
+impl ToRedisArgs for Prefix {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        "PREFIX".write_redis_args(out);
+        self.0.write_redis_args(out);
+    }
+}
+/// Redis Type: TimeoutError
+pub enum TimeoutError {
+    /// TIMEOUT
+    Timeout,
+    /// ERROR
+    Error,
+}
+
+impl ToRedisArgs for TimeoutError {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        match self {
+            TimeoutError::Timeout => "TIMEOUT".write_redis_args(out),
+            TimeoutError::Error => "ERROR".write_redis_args(out),
+        }
+    }
+}
+/// Redis Type: Operation
+pub enum Operation {
+    /// Unknown
+    Count(i64),
+    /// RESET
+    Reset,
+}
+
+impl ToRedisArgs for Operation {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        match self {
+            Operation::Count(inner) => {
+                inner.write_redis_args(out);
+            },
+            Operation::Reset => "RESET".write_redis_args(out),
+        }
+    }
+}
+/// Redis Type: Filterby
+pub enum Filterby {
+    /// MODULE
+    Module(String),
```

Comment:
> >     Do we want to allow for a better usability for the types in token.rs by adding traits like ToRedisString, ToRedisKey? These currently have underlying types based on the redis RESP (String, i64, f64)

> Adding traits how? I think this too would benefit from a review comment in context, so it's easier for me to review.

In this file in general we have types that just wrap some kind of base type. I am thinking of making this generic with a trait bound of `ToRedisString` so that I can put any type (that implements `ToRedisString`) in my code into this enum and don't  have to convert it into a `String` first. Similar for the other types (i64, f64, and probably Keys which are currently Strings too).
But that might increase compilation times a lot as there are many types here and most of them are candidates to get a generic.